src/luascript.cpp

```diff
@@ -6726,16 +6726,30 @@ int LuaScriptInterface::luaCreatureGetMaster(lua_State* L)
 
 int LuaScriptInterface::luaCreatureSetMaster(lua_State* L)
 {
-	// creature:setMaster(master)
+	// creature:setMaster(master[, force = false])
 	Creature* creature = getUserdata<Creature>(L, 1);
 	if (!creature) {
 		lua_pushnil(L);
 		return 1;
 	}
 
 	Creature* master = getCreature(L, 2);
+	bool force = getBoolean(L, 3, false);
 	if (master) {
-		pushBoolean(L, creature->convinceCreature(master));
```

Comment:
Have you tested other scripts that use this function to make sure it doesn't break it?
I tested adding the Creature::addSummon(Creature* creature) to Lua (as you pretty much did here) but that still didn't fix the bug.
Tested an older src from nightlies and had the same issue with the old C++ code.

https://github.com/otland/forgottenserver/commit/de73b47e3bd66007584a2017a7481213fc9c22af